heron/stmgr/src/cpp/manager/instance-server.cpp

```diff
@@ -154,8 +157,23 @@ InstanceServer::~InstanceServer() {
     }
   }
 
+  for (auto qlmIter = connection_buffer_length_metric_map_.begin();
+      qlmIter != connection_buffer_length_metric_map_.end(); ++qlmIter) {
+    const sp_string& instance_id = qlmIter->first;
+    for (auto iter = instance_info_.begin(); iter != instance_info_.end(); ++iter) {
```

Comment:
suggest to refactor this code into a findInstance() or findFirstInstance() function.

Reply:
simplified code in next commit